src/widgets/paragraph.rs

```diff
@@ -200,6 +202,41 @@ impl<'a> Widget for Paragraph<'a> {
     }
 }
 
+/// TODO move to a public place
+fn checked_div_ceil(value: u16, divider: u16) -> Option<u16> {
+    let v = value.checked_div(divider)?;
+    let rest = value % divider;
+
+    if rest == 0 {
+        Some(v)
+    } else {
+        Some(v + 1)
+    }
+}
+
+impl<'a> SizeHint for Paragraph<'a> {
+    fn size_hint(&self, area: &Rect) -> Size {
+        match &self.block {
+            Some(b) => {
+                let inner_area = b.inner(*area);
+                Size::new(inner_area.width, inner_area.height)
+            }
+            None => {
+                let height = self
+                    .text
+                    .lines
+                    .iter()
+                    .map(|line| {
+                        let w = line.width();
+                        checked_div_ceil(w as u16, area.width).ok_or(1u16).unwrap()
+                    })
+                    .sum();
+                Size::new(area.width, height)
+            }
+        }
+    }
+}
```

Comment:
@joshka I think I fixed this. unfortunately I had to duplicate some of the `render` code. but the size_hint calculation should work stable (since it is the same code used when rendering).

can I resolve this issue for now. maybe later we can check how to optimize things? (next step, I will try to support one or 2 other widgets)